test/connector/tcp/mssql/mssql_connector_test.go

```diff
@@ -99,3 +102,191 @@ func RunTests(t *testing.T, queryExec dbQueryExecutor) {
 	})
 
 }
+
+const mockServerPort = 2224
+type testClientParams struct {
+	queryExec dbQueryExecutor
+	applicationName string
+	serverName func(server string, port int) string
+}
+
+func TestClientParams(t *testing.T) {
+	// Setup mock-server listener
+	ln, err := net.Listen("tcp", ":1434")
+	defer func() {
+		_ = ln.Close()
+	}()
+
+	if err != nil {
+		panic(err)
+	}
+
+	sqlcmdParamsTestClient := testClientParams{
+		queryExec:        sqlcmdExec,
+		applicationName: "SQLCMD",
+		serverName: func(server string, port int) string {
+			return fmt.Sprintf(
+				"%s,%d",
+				server,
+				port,
+			)
+		},
+	}
+	gomssqlParamsTestClient := testClientParams{
+		queryExec:        gomssqlExec,
+		applicationName: "go-mssqldb",
+		serverName: func(server string, port int) string {
+			return server
+		},
+	}
+
+	t.Run(
```

Comment:
This is probably a matter of personal preference, so feel free to take it or leave it.... but it would be possible to write lines 143-190 as table-driven, e.g.:
```
	TestCases := []struct {
		description string
		readonly    bool
		testClient  testClientParams
	}{
		{
			description: "sqlcmd: client params are propagated to the server",
			readonly: false,
			testClient: sqlcmdParamsTestClient,
		},
		{
			description: "go-mssqldb: client params are propagated to the server",
			readonly: false,
			testClient: gomssqlParamsTestClient,
		},
		{
			description: "sqlcmd: readonly application intent is propagated",
			readonly: true,
			testClient: sqlcmdParamsTestClient,
		},
		{
			description: "go-mssqldb: readonly application intent is propagated",
			readonly: true,
			testClient: gomssqlParamsTestClient,
		},
	}
	for _, tc := range TestCases {
		t.Run(
			tc.description,
			func(t *testing.T) {
				propagateParams(
					t,
					tc.readonly,
					ln,
					tc.testClient,
				)
			},
		)
	}
}
```
and since there would only be one instance of call to `propagateParams()`, you could move the implementation of `propagateParams()` into the for loop above. One advantage of the table-driven approach is that you can quickly see the combinations that are being tried.